tests/text_content_detection.rs

```diff
@@ -181,88 +175,82 @@ async fn detections() -> Result<(), anyhow::Error> {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
                 contents: vec![
                     "This sentence does not have a detection.".into(),
                     "But <this one does>.".into(),
                 ],
                 detector_params: DetectorParams::new(),
-            }),
-            MockResponse::json(vec![
-                vec![],
-                vec![ContentAnalysisResponse {
-                    start: 4,
-                    end: 18,
-                    text: "this one does".into(),
-                    detection: "has_angle_brackets".into(),
-                    detection_type: "angle_brackets".into(),
-                    detector_id: Some(sentence_detector.into()),
-                    score: 1.0,
-                    evidence: None,
-                }],
-            ]),
-        ),
-    );
-
-    let mut whole_doc_detector_mocks = MockSet::new();
-    whole_doc_detector_mocks.insert(
-        MockPath::post(TEXT_CONTENTS_DETECTOR_ENDPOINT),
-        Mock::new(
-            MockRequest::json(ContentAnalysisRequest {
-                contents: vec!["This sentence has <a detection here>.".into()],
-                detector_params: DetectorParams::new(),
-            }),
-            MockResponse::json(vec![vec![ContentAnalysisResponse {
-                start: 18,
-                end: 35,
-                text: "a detection here".into(),
+            });
+        then.json(vec![
+            vec![],
+            vec![ContentAnalysisResponse {
+                start: 4,
+                end: 18,
+                text: "this one does".into(),
                 detection: "has_angle_brackets".into(),
                 detection_type: "angle_brackets".into(),
                 detector_id: Some(sentence_detector.into()),
                 score: 1.0,
                 evidence: None,
-            }]]),
-        ),
-    );
+            }],
+        ]);
+    });
+
+    let mut whole_doc_detector_mocks = MockSet::new();
+    whole_doc_detector_mocks.mock(|when, then| {
+        when.post()
+            .path(TEXT_CONTENTS_DETECTOR_ENDPOINT)
+            .json(ContentAnalysisRequest {
+                contents: vec!["This sentence has <a detection here>.".into()],
+                detector_params: DetectorParams::new(),
+            });
+        then.json(vec![vec![ContentAnalysisResponse {
+            start: 18,
+            end: 35,
+            text: "a detection here".into(),
+            detection: "has_angle_brackets".into(),
+            detection_type: "angle_brackets".into(),
+            detector_id: Some(sentence_detector.into()),
+            score: 1.0,
+            evidence: None,
+        }]]);
+    });
 
     // Start orchestrator server and its dependencies
-    let mock_chunker_server = GrpcMockServer::new(chunker_id, chunker_mocks)?;
+    let mock_chunker_server = MockServer::new(chunker_id).with_mocks(chunker_mocks);
```

Comment:
Missing `.grpc()` option
```suggestion
    let mock_chunker_server = MockServer::new(chunker_id).grpc().with_mocks(chunker_mocks);
```